StaticAnalyzer/views/match_strategy.py

```diff
@@ -0,0 +1,120 @@
+from abc import ABC, abstractclassmethod
+import re
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class MatchStrategy(ABC):
+
+    @abstractclassmethod
+    def perform_search(self, content, rule, perms):
+        """Search for instance of match in content."""
+
+
+class SingleRegex(MatchStrategy):
+
+    def perform_search(self, content, rule, perms):
+        found = False
+        if re.findall(rule['match'], content):
+            found = True
+        return found
+
+
+class RegexAnd(MatchStrategy):
+    def perform_search(self, content, rule, perms):
+        found = True
+        # This check is used because if you only pass a str rather than a list,
+        # Python will iterate through the str without raising an exception
+        if isinstance(rule['match'], str):
+            logger.debug('wrong regex type, switching to single regex')
+            return SingleRegex().perform_search(content, rule, perms)
+        for regex in rule['match']:
+            if bool(re.findall(regex, content)) is False:
+                found = False
+                break
+        return found
```

Comment:
Is the variable required? Can this be replaced with `return True`?

Reply:
Yes it can, it is more about "good practice" but as the methods are really short, I grant you that it is less relevant. it can definitely be changed.